zerver/tests/test_new_users.py

```diff
@@ -107,3 +108,16 @@ def test_notify_of_new_user_internally(self):
         self.assertEqual(actual_stream.name, 'signups')
         self.assertEqual(message.recipient.type, Recipient.STREAM)
         self.assertIn("**INTERNAL SIGNUP**", message.content)
+
+    def test_notify_realm_of_new_user(self):
+        # type: () -> None
+        new_user = self.example_user('cordelia')
+        stream = self.make_stream('announce')
+        new_user.realm.notifications_stream = stream
+        new_user.realm.save()
+        new_user = self.example_user('cordelia')
+        notify_new_user(new_user)
+
+        message = self.get_last_message()
+        actual_stream = Stream.objects.get(id=message.recipient.type_id)
+        self.assertEqual(actual_stream.name, 'announce')
```

Comment:
I merged this commit as 8cc1f13d24580222c9d927389f0de3ef59818281, after adding an extra assert that the last message was indeed to a stream.